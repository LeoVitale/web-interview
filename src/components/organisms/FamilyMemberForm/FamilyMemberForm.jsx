import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Container, Row, Col } from 'react-grid-system'
import TextBox from 'components/atoms/TextBox'
import Button from 'components/atoms/Button'

const FamilyMemberForm = ({ onSave }) => {
  const [firstName, setFirstName] = useState('')
  const [lastName, setLastName] = useState('')
  const member = {
    firstName,
    lastName,
  }
  return (
    <Container>
      <Row>
        <Col>
          <TextBox
            label="First Name"
            name="firstName"
            value={firstName}
            onChange={e => setFirstName(e.target.value)}
          />
        </Col>
        <Col>
          <TextBox
            label="Last Name"
            name="lastName"
            value={lastName}
            onChange={e => setLastName(e.target.value)}
          />
        </Col>
      </Row>
      <Row>
        <Col>
          <Button onClick={() => onSave(member)}>Save</Button>
        </Col>
      </Row>
    </Container>
  )
}

FamilyMemberForm.propTypes = {
  onSave: PropTypes.func.isRequired,
}

export default FamilyMemberForm
