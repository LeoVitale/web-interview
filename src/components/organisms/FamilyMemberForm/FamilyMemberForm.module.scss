.familyMembers {
  padding: 10px 0;
  margin: 0;
  .row {
    margin-bottom: 20px;
  }

  label {
    display: block;
    font-size: 1em;
    margin-bottom: 5px;
  }

  .select {
    font-size: 1.2em;
  }
}
