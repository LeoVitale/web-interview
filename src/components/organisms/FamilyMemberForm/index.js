export { default } from './FamilyMemberForm.jsx'
