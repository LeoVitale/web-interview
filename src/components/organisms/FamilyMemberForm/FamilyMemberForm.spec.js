import React from 'react'
import { mount } from 'enzyme'
import FamilyMemberForm from './FamilyMemberForm.jsx'

describe('<FamilyMemberForm />', () => {
  const familyMemberForm = mount(<FamilyMemberForm />)

  it('should render FamilyMemberForm as expected', () => {
    expect(familyMemberForm).toMatchSnapshot()
  })
})
