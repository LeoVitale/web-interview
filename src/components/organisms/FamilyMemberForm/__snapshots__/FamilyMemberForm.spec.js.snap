// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FamilyMemberForm /> should render FamilyMemberForm as expected 1`] = `
<FamilyMemberForm>
  <Container
    component="div"
    fluid={false}
    lg={false}
    md={false}
    sm={false}
    style={Object {}}
    xl={false}
    xs={false}
  >
    <ScreenClassResolver>
      <ScreenClassProvider
        useOwnWidth={false}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "marginLeft": "auto",
              "marginRight": "auto",
              "maxWidth": "960px",
              "paddingLeft": "15px",
              "paddingRight": "15px",
              "position": "relative",
            }
          }
        >
          <Row
            align="normal"
            component="div"
            debug={false}
            justify="start"
            nogutter={false}
            style={Object {}}
          >
            <div
              style={
                Object {
                  "alignItems": "normal",
                  "display": "flex",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "flexWrap": "wrap",
                  "justifyContent": "flex-start",
                  "marginLeft": "-15px",
                  "marginRight": "-15px",
                }
              }
            >
              <Col
                component="div"
                debug={false}
                lg={null}
                md={null}
                offset={Object {}}
                pull={Object {}}
                push={Object {}}
                sm={null}
                style={Object {}}
                xl={null}
                xs={null}
              >
                <ScreenClassResolver>
                  <div
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 0,
                        "left": "auto",
                        "marginLeft": "0%",
                        "maxWidth": "100%",
                        "minHeight": "1px",
                        "paddingLeft": "15px",
                        "paddingRight": "15px",
                        "position": "relative",
                        "right": "auto",
                        "width": "100%",
                      }
                    }
                  >
                    <TextBox
                      label="First Name"
                      name="firstName"
                      onChange={[Function]}
                      value=""
                    >
                      <div
                        className="textBox"
                      >
                        <label
                          className="title"
                          htmlFor="firstName"
                        >
                          First Name
                        </label>
                        <input
                          className="input"
                          id="firstName"
                          name="firstName"
                          onChange={[Function]}
                          type="text"
                          value=""
                        />
                      </div>
                    </TextBox>
                  </div>
                </ScreenClassResolver>
              </Col>
              <Col
                component="div"
                debug={false}
                lg={null}
                md={null}
                offset={Object {}}
                pull={Object {}}
                push={Object {}}
                sm={null}
                style={Object {}}
                xl={null}
                xs={null}
              >
                <ScreenClassResolver>
                  <div
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 0,
                        "left": "auto",
                        "marginLeft": "0%",
                        "maxWidth": "100%",
                        "minHeight": "1px",
                        "paddingLeft": "15px",
                        "paddingRight": "15px",
                        "position": "relative",
                        "right": "auto",
                        "width": "100%",
                      }
                    }
                  >
                    <TextBox
                      label="Last Name"
                      name="lastName"
                      onChange={[Function]}
                      value=""
                    >
                      <div
                        className="textBox"
                      >
                        <label
                          className="title"
                          htmlFor="lastName"
                        >
                          Last Name
                        </label>
                        <input
                          className="input"
                          id="lastName"
                          name="lastName"
                          onChange={[Function]}
                          type="text"
                          value=""
                        />
                      </div>
                    </TextBox>
                  </div>
                </ScreenClassResolver>
              </Col>
            </div>
          </Row>
          <Row
            align="normal"
            component="div"
            debug={false}
            justify="start"
            nogutter={false}
            style={Object {}}
          >
            <div
              style={
                Object {
                  "alignItems": "normal",
                  "display": "flex",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "flexWrap": "wrap",
                  "justifyContent": "flex-start",
                  "marginLeft": "-15px",
                  "marginRight": "-15px",
                }
              }
            >
              <Col
                component="div"
                debug={false}
                lg={null}
                md={null}
                offset={Object {}}
                pull={Object {}}
                push={Object {}}
                sm={null}
                style={Object {}}
                xl={null}
                xs={null}
              >
                <ScreenClassResolver>
                  <div
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 0,
                        "left": "auto",
                        "marginLeft": "0%",
                        "maxWidth": "100%",
                        "minHeight": "1px",
                        "paddingLeft": "15px",
                        "paddingRight": "15px",
                        "position": "relative",
                        "right": "auto",
                        "width": "100%",
                      }
                    }
                  >
                    <Button
                      onClick={[Function]}
                    >
                      <button
                        className="button"
                        onClick={[Function]}
                      >
                        Save
                      </button>
                    </Button>
                  </div>
                </ScreenClassResolver>
              </Col>
            </div>
          </Row>
          <span
            style={
              Object {
                "clear": "both",
                "display": "table",
              }
            }
          />
        </div>
      </ScreenClassProvider>
    </ScreenClassResolver>
  </Container>
</FamilyMemberForm>
`;
